controller/src/main/java/io/pravega/controller/server/eventProcessor/RequestHandlerMultiplexer.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor;
+
+import io.pravega.common.concurrent.FutureHelpers;
+import io.pravega.shared.controller.event.ScaleEvent;
+import io.pravega.shared.controller.event.ControllerEvent;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+@Data
+public class RequestHandlerMultiplexer implements RequestHandler<ControllerEvent> {
+    private final AutoScaleRequestHandler autoScaleRequestHandler;
+    private final ScaleOperationRequestHandler scaleOperationRequestHandler;
+
+    @Override
+    public CompletableFuture<Void> process(ControllerEvent controllerEvent) {
+        if (controllerEvent instanceof ScaleEvent) {
```

Comment:
Using visitor pattern has a problem -- we will need to add processing methods into our data objects (events) . also these events are in shared/contract because both SSS and controller post events into this stream.